aggregator/src/aggregator/problem_details.rs

```diff
@@ -36,19 +36,30 @@ pub struct ProblemDocument<'a> {
     taskid: Option<String>,
     #[serde(skip_serializing_if = "Option::is_none")]
     detail: Option<&'a str>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    collection_job_id: Option<String>,
 }
 
 impl<'a> ProblemDocument<'a> {
-    pub fn new(error_type: DapProblemType) -> Self {
+    pub fn new(type_: &'static str, title: &'static str, status: Status) -> Self {
```

Comment:
This is a breaking change to a public export from crate `janus_aggregator`. However we don't ship that crate anywhere (unlike the client and collector crates), so I'm not sure if we observe semver for this kind of thing. I bet if you look we break this crate's API all the time, but could you double check?

Reply:
If it's not in `crates.io`, as far as I'm concerned we don't have a public API to break. I think someone could manually clone the Janus repository & depend on `janus_aggregator` that way? But IMO that is not usage we want to support.

Reply:
I think that's reasonable, too. We should document this semver policy in `docs/DEVELOPMENT.md` (not needed in this PR though).

Reply:
Indeed I've been treating `aggregator`'s API boundary as its configuration and HTTP API endpoints. Agreed on writing docs, I can make a PR.

Reply:
Yep, I agree with that: a breaking change to `janus_aggregator` would be a breaking change to deployability of the artifacts (binaries/containers), rather than to the APIs in the crate.